src/lib.rs

```diff
@@ -460,13 +466,22 @@ where
                         Some(pw) => format!("redis://:{}@{}:{}", pw, host, port),
                         None => format!("redis://{}:{}", host, port),
                     },
+                    ConnectionAddr::TcpTls { ref host, port, insecure } => match &info.redis.password {
+                        Some(pw) if insecure => format!("rediss://:{}@{}:{}/#insecure", pw, host, port),
```

Comment:
```suggestion
                        Some(pw) => format!("rediss://:{}@{}:{}{}", pw, host, port, if insecure { "/#insecure" } else  { "" }),
```
Would remove two branches branches